lightningd/invoice.c

```diff
@@ -83,6 +83,7 @@ static void tell_waiter(struct command *cmd, const struct invoice *paid)
 	json_add_hex(response, "rhash", &paid->rhash, sizeof(paid->rhash));
 	json_add_u64(response, "msatoshi", paid->msatoshi);
 	json_add_bool(response, "complete", paid->state == PAID);
+	json_add_u64(response, "pay_index", paid->pay_index);
```

Comment:
Make this field conditional on 'complete' being true.  Otherwise it's just weird...